generators/app/templates/gulp/tasks/templates.js

```diff
@@ -1,12 +1,31 @@
 'use strict';
 
 var fs = require('fs');
+var path = require('path');
 
 var templatesTask = function (gulp, plugins, config, helpers) {
 
+  function getTemplateInputData(file) {
+    try {
+      var data = JSON.parse(
+        fs.readFileSync(config.src.dataPath + path.basename(file.path) + '.json')
+      );
+    } catch(e) {
```

Comment:
Also, let's get rid of the `fs` read, as it can be a perf bottleneck. We have a file buffer available under `file.contents`, so I'd go with something like `var data = JSON.parse(file.contents.toString())`